### PR TITLE
Improved Error handling Jobs

### PR DESCRIPTION
**Pull request by kensipe**
better error handling.  do not expose 404 issues to cli users.

### CODE REVIEW

File: cli/dcoscli/job/main.py [hunk 1]
```diff
@@ -446,6 +450,12 @@ def _show_schedule(job_id, json_flag=False):
     url = "{}/{}/schedules".format(_get_api_url('v1/jobs'), job_id)
     try:
         response = _do_request(url, 'GET')
+    except DCOSHTTPException as e:
+        if e.response.status_code == 404:
+            emitter.publish("Job ID: '{}' does NOT exist.".format(job_id))
```
Comment (tamarrow) [inline cli/dcoscli/job/main.py]: `raise DCOSException("Job ID: '{}' does NOT exist.".format(job_id))`

File: cli/dcoscli/job/main.py [hunk 2]
```diff
@@ -343,8 +343,12 @@ def _show(job_id):
     try:
         response = _do_request("{}/{}".format(
             _get_api_url('v1/jobs'), job_id), 'GET')
-    except DCOSException as e:
-        raise DCOSException(e)
+    except DCOSHTTPException as e:
+        if e.response.status_code == 404:
+            emitter.publish("Job ID: '{}' does not exist.".format(job_id))
```
Comment (tamarrow) [inline cli/dcoscli/job/main.py]: `raise DCOSException("Job ID: '{}' does NOT exist.".format(job_id))`

**Comment by kensipe** [comment]
@tamarrow tests added and PR feedback integrated!  ready to go!

File: cli/tests/integrations/test_job.py [hunk 3]
```diff
@@ -87,6 +87,46 @@ def test_show_job():
...
+    returncode, stdout, stderr = exec_command(
+        ['dcos', 'job', 'show', 'invalid'])
+
+    assert returncode == 1
+    assert "does NOT exist" in stderr.decode('utf-8')
+
+
+def test_show_job_runs_blank_jobname():
+    returncode, stdout, stderr = exec_command(
+        ['dcos', 'job', 'show', 'runs'])
```
Comment (tamarrow) [inline cli/tests/integrations/test_job.py]: Can we special case this to say `Job ID: '{}' does NOT exist, did you mean `dcos job show runs <job-id>`

File: cli/tests/integrations/test_job.py [hunk 4]
```diff
@@ -87,6 +87,46 @@ def test_show_job():
...
+    assert returncode == 1
+    assert stdout.decode('utf-8').startswith('Command not recognized')
+
+
+def test_show_schedule_invalid_jobname():
+    returncode, stdout, stderr = exec_command(
+        ['dcos', 'job', 'schedule', 'show', 'invalid'])
+
+    assert returncode == 1
+    assert "does NOT exist" in stderr.decode('utf-8')
```
Comment (tamarrow) [inline cli/tests/integrations/test_job.py]: We can use `assert_command` and watch the entire stderr ouput

**Comment by tamarrow** [comment]
@kensipe looks good, just a few small comments

**Comment by jsancio** [comment]
@tamarrow and @kensipe, What is the status on this? Can we merge this?

**Comment by tamarrow** [comment]
Waiting on a few pr changes before merging

**Comment by cruhland** [comment]
@tamarrow and @kensipe checking in on this again.

**Comment by kensipe** [comment]
back on this... probably done today

**Comment by kensipe** [comment]
@tamarrow this is ready to go when the test past... my changes pass locally.

**Comment by tamarrow** [comment]
thanks @kensipe! Will merge once tests pass

**Comment by tamarrow** [comment]
@kensipe 1 syntax error: `[21:59:21][Step 4/5] tests/integrations/test_job.py:98:1: E302 expected 2 blank lines, found 1`

**Comment by kensipe** [comment]
@tamarrow sorry about that.. should be fixed now.

**Comment by kensipe** [comment]
needed a rebase

**Comment by kensipe** [comment]
@tamarrow test pass ready to merge

**Comment by tamarrow** [comment]
thanks @kensipe!

**Pull request merged**